apps/worker/src/app/workflow/services/metric-queue-active.service.ts

```diff
@@ -0,0 +1,115 @@
+import { QueueService } from '@novu/application-generic';
+import { checkinForCronJob } from '../../shared/utils/cronHealth';
+import * as process from 'process';
+
+const LOG_CONTEXT = 'MetricQueueService';
+const METRIC_JOB_ID = 'metric-job';
+
+@Injectable()
+export class MetricQueueActiveService extends QueueService<Record<string, never>> {
+  constructor(@Inject('BULLMQ_LIST') private token_list: QueueService[]) {
+    super('metric-active');
+
+    this.bullMqService.createWorker(this.name, this.getWorkerProcessor(), this.getWorkerOpts());
+
+    this.bullMqService.worker.on('completed', async (job) => {
+      checkinForCronJob(process.env.ACTIVE_CRON_ID);
+      Logger.verbose('Metric job Completed', job.id, LOG_CONTEXT);
+    });
+
+    this.bullMqService.worker.on('failed', async (job, error) => {
+      Logger.verbose('Metric job failed', LOG_CONTEXT, error);
+    });
+
+    this.addToQueueIfMetricJobExists();
+  }
+
+  private addToQueueIfMetricJobExists(): void {
+    Promise.resolve(
+      this.bullMqService.queue.getRepeatableJobs().then((job): boolean => {
+        let exists = false;
+        for (const jobElement of job) {
+          if (jobElement.id === METRIC_JOB_ID) {
+            exists = true;
+          }
+        }
+
+        return exists;
+      })
+    )
+      .then(async (exists): Promise<void> => {
+        Logger.debug('metric job exists: ' + exists, LOG_CONTEXT);
+
+        if (!exists) {
+          Logger.debug(`metricJob doesn't exist, creating it`, LOG_CONTEXT);
+
+          return await this.addToQueue(METRIC_JOB_ID, {}, '', {
+            jobId: METRIC_JOB_ID,
+            repeatJobKey: METRIC_JOB_ID,
+            repeat: {
+              immediately: true,
+              pattern: '* * * * * *',
+            },
+            removeOnFail: true,
+            removeOnComplete: true,
+            attempts: 1,
+          });
+        }
+
+        return undefined;
+      })
+      .catch((error) => Logger.error('Metric Job Exists function errored', LOG_CONTEXT, error));
+  }
+
+  private getWorkerOpts(): WorkerOptions {
+    return {
+      ...this.bullConfig,
+      lockDuration: 900,
+      concurrency: 1,
+      settings: {},
+    } as WorkerOptions;
+  }
+
+  private getWorkerProcessor() {
+    return async () => {
+      return await new Promise<void>(async (resolve, reject): Promise<void> => {
+        Logger.verbose('metric job started', LOG_CONTEXT);
+        const deploymentName = process.env.FLEET_NAME ?? 'default';
+
+        try {
+          for (const queueService of this.token_list) {
+            const waitCount = await queueService.bullMqService.queue.getWaitingCount();
+            const delayedCount = await queueService.bullMqService.queue.getDelayedCount();
+            const activeCount = await queueService.bullMqService.queue.getActiveCount();
+
+            if (process.env.NOVU_MANAGED_SERVICE === 'true' && process.env.NEW_RELIC_LICENSE_KEY) {
```

Comment:
Should we validate here the environment here as well or is it redundant?

Reply:
which one are you referring to?
Here I just need to make sure we only send in the cloud version of Novu.